BUGFIX: Fixed issues related to windows filesystem using backslash instead of forward slash

UndefinedOffset opened the pull request:
When using the module on windows DIRECTORY_SEPERATOR returns a backslash instead of a forward slash. This causes one primary issue, an infinite loop when getMenu() is called causing the memory limit to be reached. It also seems to cause issues doing other tasks like adjusting the links to relative files. Issue #70 is related to and fixed by this pull request.

dhensby commented:
Good fix. `DIRECTORY_SEPARATOR` only needs to be used to parse paths from the file system, not construct them (as windows accepts both `\` and `/` for paths).

dhensby commented on code/DocumentationManifest.php:
```diff
@@ -609,6 +609,16 @@ public function normalizeUrl($url)
+     */
+    public function normalizePath($path)
+    {
+        return str_replace('\\', '/', $path);
```
I'm concerned that this may be fragile to different dir separators.

Can we use the `DIRECTORY_SEPARATOR` on this instead of hard coding `'\\'`? We could check if the dir seperator is not `/` and only then do the replacement?

UndefinedOffset replied:
Sure I'll adjust tomorrow or Monday to address it

UndefinedOffset commented:
@dhensby I've added a new commit 5b73b227688a1816bce3877b9a0315cc0a16681b that does what you requested. It also accounts for an issue I didn't spot in my testing yesterday since I was not using automatic registration. Basically the substr() check that DocumentationManifest::getRealPath() was using does not work for windows since absolute windows file system paths begin with a drive letter i.e c:\ the check for / always fails so I switched the check to Director::is_absolute() instead.

camfindlay commented:
This normalisation for paths feels like something framework should include... I'd imagine this is useful in many more places that just this module... @dhensby do you know if Framework contains such a function and if not where should this live if it we're, say, to find it's way into a Framework PR?

dhensby commented:
@camfindlay when getting the path of real files on the system, then I think this should be normalised (if they aren't already) but I think the problem here is that this isn't using the frameworks filesystem layer to find these paths, so we have to do something ourselves.

dhensby commented:
I'm happy with this. @UndefinedOffset would you mind squashing the second commit into the first one, please?

UndefinedOffset commented:
@dhensby all squashed :)

dhensby commented:
fantastic. thanks :)
